src/PRBProxy.sol

```diff
@@ -35,9 +39,39 @@ contract PRBProxy is IPRBProxy {
     }
 
     /*//////////////////////////////////////////////////////////////////////////
-                                  FALLBACK FUNCTION
+                                  FALLBACK FUNCTIONS
     //////////////////////////////////////////////////////////////////////////*/
 
+    // TODO: refactor this method and execute to use an internal execution method
+    // solhint-disable no-complex-fallback
+    fallback(bytes calldata data) external payable returns (bytes memory response) {
+        // Check if the function signature exists in the installed plugin methods mapping.
+        IPRBProxyPlugin plugin = plugins[msg.sig];
+        if (address(plugin) == address(0)) {
+            revert PRBProxy_NoPluginFound(msg.sender, msg.sig);
+        }
+
+        // Reserve some gas to ensure that the function has enough to finish the execution.
+        uint256 stipend = gasleft() - minGasReserve;
+
+        // Call method on the installed plugin.
+        bool success;
+        (success, response) = address(plugin).delegatecall{ gas: stipend }(data);
+
+        // Check if the call was successful or not.
+        if (!success) {
```

Comment:
MIssing check for owner not changed here.

Reply:
The point of the plugins is to allow other contracts (i.e., callbacks when receiving NFTs) to call the methods.

Thus, the owner should not be the only one allowed the plugin methods.

If the plugin method only accepts the owner, it should implement the restriction itself. The restriction should not be in the Proxy implementation.

Reply:
I think that you have misunderstood my comment - I was not referring to checking that the caller (`msg.sender`) is the owner. Only that *after* the delegate call, the `owner` of the proxy should remain the same. See how I check this in the `execute` function:

https://github.com/PaulRBerg/prb-proxy/blob/483affa57607c91267a6339ec5674de15f5e0069/src/PRBProxy.sol#L86-L88

Reply:
FYI, I have already implemented this on my local branch. I will push a commit later today.